controllers/storagecluster/provider_server_test.go

```diff
@@ -0,0 +1,169 @@
+package storagecluster
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v1"
+	"github.com/red-hat-storage/ocs-operator/controllers/util"
+)
+
+func TestOcsProviderServerEnsureCreated(t *testing.T) {
+
+	t.Run("Ensure that Deployment,Service,Secret is created when AllowRemoteStorageConsumers is enabled", func(t *testing.T) {
+
+		r, instance := createSetupForOcsProviderTest(t, true)
+
+		obj := &ocsProviderServer{}
+		_, err := obj.ensureCreated(r, instance)
+
+		assert.Error(t, err)
+		assert.Equal(t, "Deployment ocs-provider-server is not ready", err.Error())
+
+		deployment := &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{Name: ocsProviderServerName},
+		}
+		assertFound(t, r.Client, deployment)
+		expectedDeployment := GetProviderAPIServerDeployment(instance)
```

Comment:
Why are we testing against data coming from the code base itself? what if we have a bug in `GetProviderAPIServerDeployment`?